src/go/k8s/pkg/console/store.go

```diff
@@ -0,0 +1,107 @@
+package console
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+)
+
+// By default, Console can only be created with the same namespace as the referenced Cluster.
+// This is because Secrets, which are namespaced resource, cannot be mounted in Deployment (e.g. Schema Registry TLS).
+// Store enables creating Console in different namespace from the Cluster by saving a copy of the Secret in cache.
+
+// WANT: Make this file generic through "registering" watched resources and using dynamic client to get unstructured resources.
+
+// Store is a controller cache store for resources required across namespaces
+type Store struct {
+	cache.ThreadSafeStore
+
+	context context.Context
+	client  client.Client
+}
+
+// NewStore creates a new store
+func NewStore(cl client.Client) *Store {
+	return &Store{
+		ThreadSafeStore: cache.NewThreadSafeStore(cache.Indexers{}, cache.Indices{}),
+		context:         context.Background(),
+		client:          cl,
+	}
+}
+
+// Sync synchronizes watched resources to the store
+func (s *Store) Sync(cluster *redpandav1alpha1.Cluster) error {
```

Comment:
As I mentioned, the thing that concerns me the most is the action of copying secrets (or their content) across namespaces.

I think this is done to have a better user experience, but it creates a huge security problem in multi-tenant Kubernetes installations. Users can leverage this operator as a backdoor to read secrets from other tenants.

Reply:
We could also document what specific resources (e.g. Secrets) are "synced". The Cluster controller also currently copy Secret across namespaces [REF](https://github.com/redpanda-data/redpanda/blob/dev/src/go/k8s/pkg/resources/certmanager/type_helpers.go#L361) We might need to discuss this further, what do you guys think @alenkacz @vladoschreiner